Fixes for MinGW32 (gcc 4.8.1)

rioki opened the pull request:
Building under MinGW32 some minor things did not work. There changes address the issues.

rioki committed:
Adds <stdlib.h> to satisfy reference to abort.

rioki committed:
build: Fixes check target for mingw.

rioki committed:
test: Removes replacement snprintf for mingw.

Mingw has a proper definition of snprintf.

Nodejs-Jenkins commented:
Thank you for contributing this pull request! Here are a few pointers to make sure your submission will be considered for inclusion.

Commit rioki/libuv@dfcf504e9a33caf0bffb5251bd5ad65a63cd60c0 has the following error(s):
- Commit message must indicate the subsystem this commit changes

Commit rioki/libuv@5a892b6ce9cf16c87427524d761c81d9e2b0b501 has the following error(s):
- Commit message must indicate the subsystem this commit changes

Commit rioki/libuv@5c18dc1a3b5014f09c279b4bedb593228c4614d0 has the following error(s):
- Commit message must indicate the subsystem this commit changes

Commit rioki/libuv@b812e6ceb01d738ca7f20f8fddff4ea69c076245 has the following error(s):
- First line of commit message must be no longer than 50 characters
- Commit message must indicate the subsystem this commit changes

The following commiters were not found in the CLA:
- Sean Farrell

You can fix all these things _without_ opening another issue.

Please see [CONTRIBUTING.md](https://github.com/joyent/node/blob/master/CONTRIBUTING.md) for more information

bnoordhuis commented:
Couple of style comments but apart from that no objections from me.

@piscisaureus PTAL.

rioki commented:
Will fix the outstanding issues soon.

rioki commented:
Fixed outstanding issues. 

The last commit about swprintf was removed, mingw-w64 has the proper version. Why mingw32 "lost" the standard conforming swprintf  in the latest version is beyond me. Assuming that the standard conforming version exists is the safer route. The problem with mingw32 vs. mingw-w64 is that you can not distinguish between the two.  

Also, rebased changes against joyen/libuv/master.

bnoordhuis commented on Makefile.am:
```diff
@@ -197,6 +195,16 @@ test_run_tests_SOURCES = test/blackhole-server.c \
 
+if WINNT
+test_run_tests_SOURCES += test/runner-win.c \
+                          test/runner-wind.h                        
```
Typo?

rioki replied:
No old change, only without whitespace change, or am I missing something? All changes are rebased, so the history looks nice.

bnoordhuis replied:
I mean that it's test/runner-win.h, not test/runner-wind.h. :-)

rioki commented:
Fixes typo in Makefile.am, rebase from joyen/libuv/master.

bnoordhuis commented:
Thanks, LGTM.  Can I ask you to sign [the CLA](http://nodejs.org/cla.html)?

I don't think we'll wait for @piscisaureus's review if we want to see this land this side of the Singularity. :-)

rioki commented:
singed, real name is Sean Farrell (if that was not obvious from the commit...)

bnoordhuis commented:
Cheers Sean, landed in ee434b3, 4c6294e and 63cd99e.
